### PR TITLE
Update to use zend-diactoros 1.8.0

### PR DESCRIPTION
**Pull request by weierophinney**
This patch updates the component to use zend-diactoros 1.8.0 and later. In particular, it updates the `ServerRequestSwooleFactory` to use the new standalone functions in Diactoros to generate `ServerRequest` artifacts before instantiating the request itself. It also modifies the logic to inject cookies _afterwards_ from the `Cookie` header if `swoole_http_request->cookie` is empty.

It also imports and updates `SwooleStream` to ensure it follows all PSR-7 guidelines, including ensuring that `getContents()` only returns from the current index.

My main questions/concerns at this point are:

- Does swoole file uploads for _all_ request methods? If so, this would likely alleviate most of the remaining concerns I have.
- If it does not, memoizing the raw content in a property could deplete memory. Should SwooleStream instead create a `Zend\Diactoros\Stream` and proxy to it for all methods that operate on the content (`seek`, `tell`, `read`, `getContents`, `getSize`, `eof`, `__toString`, etc.)?
- Even if it does, would the above make sense when dealing with large request bodies?

### CODE REVIEW

File: src/Container/ServerRequestSwooleFactory.php [hunk 1]
```diff
@@ -21,7 +28,35 @@ class ServerRequestSwooleFactory
     public function __invoke(ContainerInterface $container) : callable
     {
         return function (swoole_http_request $request) {
-            return ServerRequestFactory::fromSwoole($request);
+            // Aggregate values from Swoole request object
+            $get     = $request->get ?? [];
+            $post    = $request->post ?? [];
+            $cookie  = $request->cookie ?? [];
+            $files   = $request->files ?? [];
+            $server  = $request->server ?? [];
```
Comment (ezimuel) [inline src/Container/ServerRequestSwooleFactory.php]: We don't need anymore to convert the server key values in uppercase?

File: src/Container/ServerRequestSwooleFactory.php [hunk 2]
```diff
@@ -21,7 +28,35 @@ class ServerRequestSwooleFactory
...
+            $headers = $request->header ?? [];
```
Comment (ezimuel) [inline src/Container/ServerRequestSwooleFactory.php]: We don't need anymore to convert `-` with `_` as reported in the previous code (as follows)?

```php
$headers = [];
if (isset($request->header)) {
    foreach ($request->header as $name => $value) {
        $headers[str_replace('-', '_', $name)] = $value;
    }
}
```

File: src/Container/ServerRequestSwooleFactory.php (hunk 1 above)
Comment (weierophinney) [reply src/Container/ServerRequestSwooleFactory.php]: The previous version wasn't converting them to uppercase, from what I could see. We can make that explicit, though, using `array_change_key_case($server, CASE_UPPER)`, though. I'm not sure what casing Swoole uses...

File: src/Container/ServerRequestSwooleFactory.php (hunk 2 above)
Comment (weierophinney) [reply src/Container/ServerRequestSwooleFactory.php]: No - we shouldn't do that. Headers are typically `-` separated, not `_` separated, and converting them to `_` will cause lookups to fail.

File: src/Container/ServerRequestSwooleFactory.php (hunk 1 above)
Comment (weierophinney) [reply src/Container/ServerRequestSwooleFactory.php]: I've updated to make this explicit casting in the `$server` array.

File: src/Container/ServerRequestSwooleFactory.php (hunk 2 above)
Comment (weierophinney) [reply src/Container/ServerRequestSwooleFactory.php]: Verified this behavior in both Swoole versions 2.2 and 4.0.1.

**Comment by weierophinney** [comment]
I've verified that cookies are _always_ parsed by `Swoole\Http\Server`, so there's no need to parse the cookie header ourselves.

**Comment by weierophinney** [comment]
I received feedback from the Swoole team on how to test the HTTP server so as to prevent blocking issues when we have multiple servers fired up during testing. The trick is to wrap the server within a child process, and message artifacts to the parent process, and then do tests from there. This approach works across both Swoole 2.2 and 4.0.1, allowing us to recommend either version to our users.

**Comment by weierophinney** [comment]
> My main questions/concerns at this point are:
> 
> - Does swoole file uploads for all request methods? If so, this would likely alleviate most of the remaining concerns I have.
> - If it does not, memoizing the raw content in a property could deplete memory. Should SwooleStream instead create a Zend\Diactoros\Stream and proxy to it for all methods that operate on the content (seek, tell, read, getContents, getSize, eof, __toString, etc.)?
> - Even if it does, would the above make sense when dealing with large request bodies?

Regarding these points:

- Yes, as with cookies and headers, the Swoole HTTP server appears to ALWAYS parse uploaded files into a structure resembling `$_FILES`.
- We reached out to Dmitry on the PHP internals team regarding string variable semantics in PHP 7 versions. As it turns out, strings are now copy-on-write, which means if we do not _modify_ the string returned by `$request->rawContent()`, we will not incur any additional memory overhead. This is precisely the current approach in the `SwooleStream` at this time, which makes it safe.

**Pull request merged**